src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -193,15 +193,17 @@ public function remove(string $key)
     /**
      * Returns the HTTP header value converted to a date.
      *
+     * @return \DateTimeImmutable|null
+     *
      * @throws \RuntimeException When the HTTP header is not parseable
      */
-    public function getDate(string $key, \DateTime $default = null): ?\DateTimeInterface
+    public function getDate(string $key, \DateTimeInterface $default = null): ?\DateTimeInterface
     {
         if (null === $value = $this->get($key)) {
             return $default;
```

Comment:
If we want to enforce returning a DateTimeImmutable, we need to convert the default